pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -122,12 +121,16 @@ public int[][] transformToDictIdsMV(ValueBlock valueBlock) {
     throw new UnsupportedOperationException();
   }
 
+  protected void initIntValuesSV(int length) {
+    if (_intValuesSV == null || _intValuesSV.length < length) {
```

Comment:
Do we need to do anything if _intValuesSV.length > length?

Reply:
No because we use it as a reusable buffer to reduce the allocation. We'll only read up to `length` values from the buffer